plugins/@grouparoo/airtable/src/lib/destination/exportRecords.ts

```diff
@@ -0,0 +1,244 @@
+              return;
+            }
+            const found = getByForeignKey(key.toString());
+            if (found) {
+              found.destinationId = record.id;
+            }
+          });
+        })
+        .catch((reason: AirtableError) => {
+          attachErrorsToErroneousRecords(users, reason);
+        });
+    }
+  };
+
+// fetch using the keys to set destinationId and result on BatchExports
+// use the getByForeignKey to lookup results
+const findAndSetDestinationIds: BatchMethodFindAndSetDestinationIds = async (
+  options: FindAndSetMethodOptions
+) => {
+  const { config, client, getByForeignKey, foreignKeys } = options;
+  const { tableId, primaryKey } =
+    config.destinationOptions as AirtableDestinationOptions;
+  const records = await client.listRecordsByField(
+    tableId,
+    primaryKey,
+    foreignKeys
+  );
+  records.forEach((record) => {
+    const key = record.fields[primaryKey.toString()];
+    if (!key) {
+      return;
+    }
+    const found = getByForeignKey(key.toString());
+    if (found) {
+      found.destinationId = record.id;
+      found.result = record.fields;
+    }
+  });
+};
+
+function attachErrorsToErroneousRecords(
+  records: BatchExport[],
+  error: AirtableError
+) {
+  records.map((record) => {
+    record.error = error;
+  });
+}
+
+function buildUpdatePayload(
+  exportedProfile: BatchExport
+): RecordData<Partial<FieldSet>> {
+  if (!exportedProfile.destinationId) {
+    throw ReferenceError(
+      `Could not find destination ID in Batch Export with key ${exportedProfile.recordId}`
+    );
+  }
+  return {
+    id: exportedProfile.destinationId,
+    fields: exportedProfile.newRecordProperties,
+  };
+}
+
+function buildCreatePayload(
+  exportedProfile: BatchExport
+): CreateRecord<FieldSet> {
+  return {
+    fields: exportedProfile.newRecordProperties,
+  };
+}
+
+const addToGroups: BatchMethodAddToGroups = async () => {
+  // This destination doesn't do groups
+};
+
+const removeFromGroups: BatchMethodRemoveFromGroups = async () => {
+  // This destination doesn't do groups
+};
+// mess with the keys (lowercase emails, for example)
+const normalizeForeignKeyValue: BatchMethodNormalizeForeignKeyValue = ({
+  keyValue,
+}) => {
+  if (!keyValue) {
+    return null;
+  }
+  return keyValue.toString().toLowerCase().trim();
+};
+
+// mess with the names of groups (tags with no spaces, for example)
+const normalizeGroupName: BatchMethodNormalizeGroupName = ({ groupName }) => {
+  return groupName.toString().trim();
+};
+interface ExportBatchOptions {
+  appOptions: SimpleAppOptions;
+  destinationOptions: SimpleDestinationOptions;
+  syncOperations?: DestinationSyncOperations;
+  exports: BatchExport[];
+}
+export async function exportBatch(exportBatchOptions: ExportBatchOptions) {
+  const { syncOperations, appOptions, destinationOptions, exports } =
+    exportBatchOptions;
+  const { primaryKey } = destinationOptions;
+  const batchSize = 10;
+  const findSize = 100;
```

Comment:
Where did these constant values come from?

Reply:
Added a constant for these values. They should both be 10 for now since that is the amount of records that the Airtable Team recommends for batching requests